third_party/move/move-compiler-v2/src/pipeline/ability_checker.rs

```diff
@@ -78,10 +78,31 @@ fn check_drop_for_temp_with_msg(
     check_drop(func_target, ty, loc, err_msg)
 }
 
+/// If temporary variable `t` does not have ability `drop`, generates an error,
+/// unless the state before `code_offset` won't return.
+fn cond_check_drop_for_temp_with_msg(
+    func_target: &FunctionTarget,
+    code_offset: CodeOffset,
+    t: TempIndex,
+    loc: &Loc,
+    err_msg: &str,
+) {
+    let abort_state = get_abort_state_at(func_target, code_offset);
+    if abort_state.before.may_return() {
+        check_drop_for_temp_with_msg(func_target, t, loc, err_msg)
```

Comment:
Given this is the only use of this function, maybe inline the contents of the call here, and remove the newly added `cond_` prefix to this function name?

Reply:
I was assuming that the rust compiler could do the inline.

Reply:
This is not an issue of performance, but having unnecessary abstractions.

We have a function `check_drop_for_temp_with_msg` that is only used once and only in a "conditional version" of that function: `cond_check_drop_for_temp_with_msg`.  The `cond_check_drop_for_temp_with_msg` is the version that is used everywhere, so it would be good to not have `check_drop_for_temp_with_msg` as a separate function, but contents inlined into its conditional version. In this manner, in future changes, we would not accidentally use the unconditional version.

Reply:
done